daml-assistant/src/DA/Daml/Assistant/Install/Path.hs

```diff
@@ -76,13 +79,49 @@ regQueryStringValue key valueName =
             failUnlessSuccess "regQueryDefaultValue" $
                 c_RegQueryValueEx p_key c_valueName nullPtr nullPtr c_value p_valueLen
             peekTString (castPtr c_value)
-#else
-updatePath _ output targetPath = suggestAddToPath output targetPath
-#endif
 
 suggestAddToPath :: (String -> IO ()) -> FilePath -> IO ()
 suggestAddToPath output targetPath = do
     -- Ask user to add .daml/bin to PATH if it is absent.
     searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
     when (targetPath `notElem` searchPaths) $ do
         output ("Please add " <> targetPath <> " to your PATH.")
+#else
+updatePath _ output targetPath = do
+  searchPaths <- map dropTrailingPathSeparator <$> getSearchPath
+  unless (targetPath `elem` searchPaths) $ do
+    shellPath <- getEnv "SHELL"
+    let shell = takeFileName shellPath
+    case shellConfig shell targetPath of
+      Nothing -> output setPathManualMsg
+      Just (configFile, cmd) -> do
+        home <- getHomeDirectory
+        let cfgFile = home </> configFile
+        cfgFileExists <- doesFileExist cfgFile
+        if cfgFileExists
+          then do
+            content <- readFile cfgFile
+            unless (cmd `elem` lines content) $ do
+              answer <- prompt "Add DAML Assistant executable to your PATH?" "Yes" ["No"]
+              when (answer `elem` ["Yes", "yes", "y", "Y"]) $ appendFile cfgFile cmd
+          else output setPathManualMsg
+  where
+    setPathManualMsg = "Please add " <> targetPath <> " to your PATH."
```

Comment:
I think we should also check `hIsTerminalDevice stdin` and `hIsTerminalDevice stdout` to make sure we're using an interactive shell, otherwise we shouldn't be doing interactive prompts like this. Otherwise I'm afraid this PR would break existing shell scripts (and other integrations).

Reply:
good point, I'll add a check.